crates/sui-core/src/signature_verifier.rs

```diff
@@ -92,12 +91,12 @@ pub struct SignatureVerifier {
     certificate_cache: VerifiedDigestCache<CertificateDigest>,
     signed_data_cache: VerifiedDigestCache<SenderSignedDataDigest>,
 
-    /// Map from kid (key id) to the fetched OAuthProviderContent for that key.
+    /// Map from (kid, iss) to the fetched JWK for that key.
     /// We use an immutable data structure because verification of ZKLogins may be slow, so we
     /// don't want to pass a reference to the map to the verify method, since that would lead to a
     /// lengthy critical section. Instead, we use an immutable data structure which can be cloned
     /// very cheaply.
-    oauth_provider_jwk: RwLock<ImHashMap<String, OAuthProviderContent>>,
+    oauth_provider_jwk: RwLock<HashMap<(String, String), JWK>>,
```

Comment:
does this have any implications to be forward/backward compatibility? @mystenmark

Reply:
reverted back to ImHashMap for now